src/charm.py

```diff
@@ -90,13 +90,17 @@ def __init__(self, *args: typing.Any):
         """
         super().__init__(*args)
         self.haproxy_service = HAProxyService()
+        # Order is important here as we want _ensure_tls to check if the hostname is configured
+        self.framework.observe(self.on[TLS_CERT_RELATION].relation_created, self._ensure_tls)
+        self.framework.observe(self.on[TLS_CERT_RELATION].relation_changed, self._ensure_tls)
```

Comment:
Is it ok if is called twice? After relation created, a relation changed is emitted too.

Reply:
I believe that's fine since the handler is only checking if the charm is ready to handle TLS ( and defer it if that's not the case)